pkg/cloudprovider/instancetypes.go

```diff
@@ -99,20 +101,27 @@ func (p *InstanceTypeProvider) List(ctx context.Context, kc *v1alpha5.KubeletCon
 	if err != nil {
 		return nil, err
 	}
-	var result []*cloudprovider.InstanceType
 
+	// Compute fully initialized instance types hash key
+	instanceTypeZonesHash, _ := hashstructure.Hash(instanceTypeZones, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
+	kcHash, _ := hashstructure.Hash(kc, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
+	key := fmt.Sprintf("%d-%d-%s-%016x-%016x", p.instanceTypesSeqNum, p.unavailableOfferings.SeqNum, nodeTemplate.UID, instanceTypeZonesHash, kcHash)
+
+	if item, ok := p.cache.Get(key); ok {
+		return item.([]*cloudprovider.InstanceType), nil
+	}
+
+	var result []*cloudprovider.InstanceType
 	for _, i := range instanceTypes {
 		instanceTypeName := aws.StringValue(i.InstanceType)
 		instanceType := NewInstanceType(ctx, i, kc, p.region, nodeTemplate, p.createOfferings(ctx, i, instanceTypeZones[instanceTypeName]))
 		result = append(result, instanceType)
 	}
+	p.cache.SetDefault(key, result)
 	return result, nil
 }
 
 func (p *InstanceTypeProvider) LivenessProbe(req *http.Request) error {
-	p.Lock()
-	//nolint: staticcheck
```

Comment:
Are we even running these probes anymore?

Reply:
yeah, looks like it https://github.com/aws/karpenter/blob/main/cmd/controller/main.go#L48